src/CommandingOfficers/CommanderBear_Bull.java

```diff
@@ -0,0 +1,180 @@
+{
+  private static final CommanderInfo coInfo = new CommanderInfo("Bear&Bull", CommanderLibrary.CommanderEnum.BEAR_BULL);
+  private final int incomeBase;
+  public boolean isBull;
+
+  private final double BEAR_MOD = 0.9;
+  private final double BULL_MOD = 1.2;
+
+  public CommanderBear_Bull()
+  {
+    super(coInfo);
+
+    incomeBase = incomePerCity;
+    // we start in Bear mode, so swap to it at the start
+    isBull = true;
+    swapD2Ds(true);
+
+    addCommanderAbility(new UpDownTurnAbility(this));
+    addCommanderAbility(new BustBoomAbility(this));
+  }
+
+  public void swapD2Ds(boolean setIncome)
+  {
+    if( isBull )
+    {
+      isBull = false;
+      if( setIncome )
+        incomePerCity = (int) (incomeBase * BEAR_MOD);
+      for( UnitModel um : unitModels )
+      {
+        um.COcost = BEAR_MOD;
+      }
+    }
+    else
+    {
+      isBull = true;
+      if( setIncome )
+        incomePerCity = (int) (incomeBase * BULL_MOD);
+      for( UnitModel um : unitModels )
+      {
+        um.COcost = BULL_MOD;
+      }
+    }
+  }
+
+  /**
+   * Down/UpTurn swaps D2Ds for this turn.
+   * All units on a property you own take 3HP mass damage and you gain the funds value of that HP.
+   * Bear: Gives you a fast injection of funds based on 1.2x your liquidated units' HP value. Should probably build before using.
+   * Bull: Having units on properties when you use it is poor due to the 0.9x liquidation price, however, the discount should be welcome.
+   */
+  private static class UpDownTurnAbility extends CommanderAbility implements COModifier
+  {
+    private static final String UPTURN_NAME = "UpTurn";
+    private static final String DOWNTURN_NAME = "DownTurn";
+    private static final int DOWNUPTURN_COST = 3;
+    private static final int DOWNUPTURN_LIQUIDATION = -3;
+    CommanderBear_Bull COcast;
+
+    UpDownTurnAbility(Commander commander)
+    {
+      // as we start in Bear form, UpTurn is the correct starting name
+      super(commander, UPTURN_NAME, DOWNUPTURN_COST);
+      COcast = (CommanderBear_Bull) commander;
+    }
+
+    @Override
+    public String toString()
+    {
+      myName = (COcast.isBull) ? DOWNTURN_NAME : UPTURN_NAME;
+      return myName;
+    }
+
+    @Override
+    protected void perform(GameMap gameMap)
+    {
+      myCommander.addCOModifier(this);
+    }
+
+    @Override // COModifier interface.
+    public void apply(Commander commander)
+    {
+      CommanderBear_Bull cmdr = (CommanderBear_Bull) commander;
+      cmdr.swapD2Ds(false);
+      // Damage is dealt after swapping D2Ds so it's actually useful to Bear
+      for( Location loc : cmdr.ownedProperties )
+      {
+        if( loc.getOwner() == myCommander )
+        {
+          Unit victim = loc.getResident();
+          if( null != victim )
+          {
+            double delta = victim.alterHP(DOWNUPTURN_LIQUIDATION);
+            cmdr.money += (-1 * delta * victim.model.getCost()) / 10;
```

Comment:
I might move the `-1` up a line, and/or drop a comment to clarify that `alterHP` is expected to return a negative value.

Reply:
Hokay.